File: ticket-frontend/src/modules/movie/dtos/movie-list-item-dto.ts
import { TypeUtils } from '../../core/type-utils';

export class MovieListItemDto {
    movieId: number = 0;
    name: string = '';
    description: string | null = null;
    image: string = '';
    imageName: string = '';
    imageMimePrefix: string = '';
    year: number = 0;

    static fromObj(obj: any): MovieListItemDto | null {
        if (!obj) {
            return null;
        }
        const movieModel: MovieListItemDto = new MovieListItemDto();
        movieModel.movieId = obj.movieId;
        movieModel.name = obj.name;
        movieModel.description = obj.description;
        movieModel.image = obj.image;
        movieModel.imageName = obj.imageName;
        movieModel.imageMimePrefix = obj.imageMimePrefix;
        movieModel.year = obj.year;
        return movieModel;
    }

    static listFromObjList(objs: any[]): MovieListItemDto[] {
        if (!objs) {
            return [];
        }
        return objs.map(MovieListItemDto.fromObj).filter(TypeUtils.isNonNullable);
    }
}

File: ticket-backend/src/main/java/aics/domain/movie/dtos/MovieListItemDto.java
package aics.domain.movie.dtos;

import aics.domain.movie.entities.Movie;
import lombok.Data;
import lombok.experimental.Accessors;

import java.io.Serializable;
import java.util.Base64;

@Data
@Accessors(chain = true)
public class MovieListItemDto implements Serializable {
    private Long movieId;
    private String name;
    private String description;
    private String image;
    private String imageName;
    private String imageMimePrefix;
    private int year;

    public static MovieListItemDto fromMovie(Movie movie) {
        if (movie == null) {
            return null;
        }
        return new MovieListItemDto()
            .setMovieId(movie.getMovieId())
            .setName(movie.getName())
            .setDescription(movie.getDescription())
            .setImage(Base64.getEncoder().encodeToString(movie.getImage()))
            .setImageName(movie.getImageName())
            .setImageMimePrefix(movie.getImageMimePrefix())
            .setYear(movie.getYear());
    }
}
